### PR TITLE
Skip no-op checkouts in traverse

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

**Review by mkondratek — APPROVED** [review]
some questions but apppppppproving

File: frontend/resourcebundles/src/main/resources/GitMacheteBundle.properties [hunk 1]
```diff
@@ -297,7 +297,7 @@ string.GitMachete.BranchOrCommitCellRendererComponent.sync-to-parent-status-tool
 string.GitMachete.BranchOrCommitCellRendererComponent.sync-to-parent-status-tooltip.in-sync.HTML=<html>Branch ''{0}'' is in sync to ''{1}''</html>
 string.GitMachete.BranchOrCommitCellRendererComponent.sync-to-parent-status-tooltip.in-sync-but-fork-point-off.HTML=<html>Branch ''{0}'' is in sync to ''{1}'', but the range of commits belonging to ''{0}'' is uncertain.<br>Consider overriding fork point from right-click menu or reattaching ''{0}'' under a different parent branch</html>
 string.GitMachete.BranchOrCommitCellRendererComponent.sync-to-parent-status-tooltip.merged-to-parent.HTML=<html>Branch ''{0}'' is merged to ''{1}''.<br>Consider sliding it out from right-click menu</html>
-string.GitMachete.BranchOrCommitCellRendererComponent.sync-to-parent-status-tooltip.out-of-sync.HTML=<html>Branch ''{0}'' is out of sync to ''{1}''.<br>Consider syncing by rebase or merge to ''{1}'' from right-click menu</html>
+string.GitMachete.BranchOrCommitCellRendererComponent.sync-to-parent-status-tooltip.out-of-sync.HTML=<html>Branch ''{0}'' is out of sync to ''{1}''.<br>Consider syncing by rebase to ''{1}'' from right-click menu</html>
```
[the remark this answers is not in the document]
Comment (PawelLipski) [reply frontend/resourcebundles/src/main/resources/GitMacheteBundle.properties:300]: Right, fixed on PR #1413

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java [hunk 2]
```diff
@@ -83,35 +88,51 @@ public void execute() {
     }
 
     switch (syncToRemoteStatus) {
+      case NoRemotes :
+      case InSyncToRemote :
+        ModalityUiUtil.invokeLaterIfNeeded(ModalityState.NON_MODAL, traverseNextEntry);
```
[the remark this answers is not in the document]
Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java:93]: :yes:, IIUC a refresh isn't needed. Each side-effecting action like push/rebase is responsible for refreshing repository on its own, so we can assume that the repository is already up to date once we enter `void execute()`.

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToParent.java [hunk 3]
```diff
@@ -43,31 +46,40 @@ public TraverseSyncToParent(GitRepository gitRepository,
...
-  @UIEffect
   public void execute() {
-    @UI Runnable syncToRemoteRunnable = new TraverseSyncToRemote(gitRepository, graphTable, gitMacheteBranch,
+    Runnable syncToRemoteRunnable = new TraverseSyncToRemote(gitRepository, graphTable, gitMacheteBranch,
         traverseNextEntry)::execute;
 
     val syncToParentStatus = gitMacheteBranch.getSyncToParentStatus();
     switch (syncToParentStatus) {
+      case InSync :
+        ModalityUiUtil.invokeLaterIfNeeded(ModalityState.NON_MODAL, syncToRemoteRunnable);
```
[the remark this answers is not in the document]
Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToParent.java:56]: Lemme explain that in a comment for both here and `NoRemotes`/`InSyncToRemote` cases

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java:93]: awesome 🚀

**Pull request merged**